### PR TITLE
Editorial review

### PR DESCRIPTION
**Pull request by aniluede**
Hi there, 

aiming to contribute an editorial review. 

As a first step, added TODO comments with questions about the documentation. Kindly have a look.

### CODE REVIEW

**Comment by aniluede** [comment]
I have read the CLA Document and I hereby sign the CLA

File: documentation/modules/rql/pages/index.adoc [hunk 1]
```diff
@@ -1,9 +1,18 @@
 = Bosch Semantic Stack RQL
+//TODO ? How come we mention "Bosch Semantic Stack" in an open-source project? Or do we want to emphasize here that this is made for Bosch Semantic Stack just that it is also open-source? (I suppose, yes. As opposed to ESMF open-source components which highly value their independence from anything commercial or Boschy.)
 
-Fundamentally, RQL (Resource Query Language) serves as a query language tailored for use in REST, specifically within URIs featuring data structures resembling objects.
+Fundamentally, RQL (Resource Query Language) serves as a query language tailored for use with REST APIs, specifically within URIs featuring data structures that resemble objects.
+//TODO ? This is "the RQL" which specification sits on GitHub under persvr/rql, right? We might add a link.
 
-The main parts of the RQL are: select, filter and option.
+The main parts of RQL are: select, filter, and option.
+//TODO ? "Parts" means functions? Features? Use cases? Applications?
```
Comment (aniluede) [inline documentation/modules/rql/pages/index.adoc:8]: parts of the RQL query

File: documentation/modules/rql/pages/index.adoc [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+//TODO (i) Aiming to rewrite towards sth like: "The main use cases for RQL are to select data, to filter data, and to ... (option?) data."
+
+//TODO ! Missing information. Why does Bosch Semantic Stack RQL exist? (To ease usage of the Digital Twin Registry.) For whom is it made? (Anybody who has digital twins or AAS in the DTR and creates queries to access them. Application developers.) Why is it open-source? (To invite users to actively work with and extend Bosch Semantic Stack RQL where they see fit (because being extensible seems to be a core feature of RQL?).) Who is behind it/maintaining it? (Bosch.)
 
 Further information can be found in the xref:rql:user-guide.adoc[User Guide].
+//TODO ? Information for whom? When it comes to shaping queries, why divide between users and developers? Aren't developers THE users here?
 
-The xref:rql:developer-guide.adoc[Developer Guide] helps you to get up-and-running with the RQL Project and how you can use it in your projects.
\ No newline at end of file
+The xref:rql:developer-guide.adoc[Developer Guide] helps you to get up-and-running with the RQL Project and details how you can use it in your projects.
```
Comment (aniluede) [inline documentation/modules/rql/pages/index.adoc:17]: not THe Project -- rather: include library

File: documentation/modules/rql/pages/user-guide.adoc [hunk 3]
```diff
@@ -114,59 +120,73 @@ option=limit(_start_,_count_),sort(-+attribute1,attribute2,attributeN)
...
-Sorts the data based on the specified attribute in the order indicated by the prefix (+ for ascending, - for descending) The attributes names are separated by a comma.
+Sorts the data based on the specified attribute in the order indicated by the prefix (+ for ascending, - for descending). The attribute names are separated by a comma.
+
 With user-defined sorting, the client is responsible for preventing an arbitrary order of the retrieved results.
 
 [subs="quotes"]
 ----
 option=sort(-+attribute1,attribute2,attributeN)
 ----
+//TODO ? Why -+? Wouldn'T it suffice to only add + and the attribute names? Is it required to always note down - and +? Does the order matter?
```
Comment (MelleD) [inline documentation/modules/rql/pages/user-guide.adoc:131]: option=sort(-attribute1,+attribute2,[+-]attributeN)

Yes order is important

File: documentation/modules/rql/pages/user-guide.adoc [hunk 4]
```diff
@@ -114,59 +120,73 @@ option=limit(_start_,_count_),sort(-+attribute1,attribute2,attributeN)
...
+
+Currently, the syntax is related to offset pagination.
+Other strategies are currently not provided but may be added in the future, e.g., cursor-based pagination.
+//TODO ? Users might want to know: What even is offset pagination?
 
 [subs="quotes"]
 ----
 options=limit(_start_,_count_)
 ----
+//TODO ? And I don't have to add numbers for start or count? I just literally add it like that to the query?
```
Comment (MelleD) [inline documentation/modules/rql/pages/user-guide.adoc:145]: Yes numbers :)

File: documentation/modules/rql/pages/user-guide.adoc [hunk 5]
```diff
@@ -114,59 +120,73 @@ option=limit(_start_,_count_),sort(-+attribute1,attribute2,attributeN)
...
+//TODO ? Isn't this rather indicating the allowed characters whereas the syntax comes in the examples below?
 
 *Examples:*
 
 name:: simple attribute named 'name'
+
 id:: simple attribute named 'name'
-location_v_2:: simple attribute named containing underscore and numbers
-_someStrangeThing:: attributes _may_ start with underscore but why would you?
+//TODO ? id ... named 'name'?
```
Comment (MelleD) [inline documentation/modules/rql/pages/user-guide.adoc:158]: correct

File: documentation/modules/rql/pages/user-guide.adoc [hunk 6]
```diff
@@ -114,59 +120,73 @@ option=limit(_start_,_count_),sort(-+attribute1,attribute2,attributeN)
...
 address/street:: _nested_ attribute 'street' of parent attribute 'address'
 
-99NotValid:: *invalid* - attributes may not start with numbers
-/alsoNotValid:: *invalid* - attributes may not start with nesting separator
-willBeParsedButHasNoMeaning/:: *no meaning* - the parser will read this but it makes no sense and *may be removed in future releases*
+99NotValid:: *invalid* -- attributes must not start with numbers
+
+/alsoNotValid:: *invalid* -- attributes may not start with nesting separator
+
+willBeParsedButHasNoMeaning/:: *no meaning* -- the parser will read this but it makes no sense and *may be removed in future releases*
```
Comment (MelleD) [inline documentation/modules/rql/pages/user-guide.adoc:171]: Please remove it

File: documentation/modules/rql/pages/user-guide.adoc [hunk 7]
```diff
@@ -189,30 +209,36 @@ Escape-char is backslash `\`, possible escape sequences are
 `\"`:: Double Quote
 `\f`:: Formfeed - *DO NOT USE* _this might be removed in future_
 `\b`:: Backspace - *DO NOT USE* _this might be removed in future_
+//TODO ? What do they do, why would I add them to a query?
```
Comment (aniluede) [inline documentation/modules/rql/pages/user-guide.adoc:212]: remove

Comment (MelleD) [inline documentation/modules/rql/pages/user-guide.adoc:212]: Remove the example which will be removed :)

File: documentation/modules/rql/pages/user-guide.adoc [hunk 8]
```diff
@@ -189,30 +209,36 @@ Escape-char is backslash `\`, possible escape sequences are
...
+So `2007-12-03T10:15Z` and `2007-12-03T10:15+4:27` are valid _ISO8601_ but *not* valid RQL.
+//TODO ? Like generally not valid for RQL or not valid within this Bosch Semantic Stack RQL project? (Also, just noting that so far it was not explained how to actually set up the RQL project so that it can be used. Or is there nothing to set up because the DTR supports it just natively? (Then why is it an extra open-source project and not just a DTR feature? Can it also be used for other registries or even any other API?))
 
 Valid examples are `2007-12-03T10:15:30.0123Z` and `2007-12-03t10:15:30+04:37`.
 
 <<<
+//TODO ?
 
 === Set comparisons
+//TODO ? Means comparing data SETs? Or to set a comparison? Or that there are some comparisons that have been set by default? Lost here. What about an intro sentence and one or two examples like above?
```
Comment (aniluede) [inline documentation/modules/rql/pages/user-guide.adoc:232]: array of values, a set of values

File: documentation/modules/rql/pages/user-guide.adoc [hunk 9]
```diff
@@ -189,30 +209,36 @@ Escape-char is backslash `\`, possible escape sequences are
...
 
 Wildcards for the like operators are `?` for a single undefined character and `*` for any number of characters.
 
 === Simple comparisons
+//TODO ? How do simple comparisons differ from "set comparisons"?
 
 <1> Supports all literals
 <2> Does not support booleans or null, as less/greater are not defined independent of implementation
 <3> Supports only strings
+//TODO ? Why formatting the numbers as if they described something like a legend for a chart or map? Is this just a bulleted list? What does it even mean, this list? An intro sentence would help.
```
Comment (aniluede) [inline documentation/modules/rql/pages/user-guide.adoc:244]: Keep <2>

File: documentation/modules/rql/pages/user-guide.adoc (hunk 8 above)
Comment (aniluede) [reply documentation/modules/rql/pages/user-guide.adoc:232]: camparison chapter -- new column in table see below

File: documentation/modules/rql/pages/developer-guide.adoc [hunk 10]
```diff
@@ -53,26 +59,27 @@ To create a string representation of the model, we can use the _toString()_ meth
 
 ===== Single argument or body
 
-Do you transport the query via html body or as a single query attribute?
-To parse a query from Rest:
+Do you transport the query via HTML body or as a single query attribute?
```
Comment (aniluede) [inline documentation/modules/rql/pages/developer-guide.adoc:62]: html-->http

Comment (MelleD) [inline documentation/modules/rql/pages/developer-guide.adoc:62]: HTML should be http

File: documentation/modules/rql/pages/developer-guide.adoc [hunk 11]
```diff
@@ -53,26 +59,27 @@ To create a string representation of the model, we can use the _toString()_ meth
...
 
 ----
 
-<1> the whole query taken from the html-body or a single query argument.
+<1> The whole query taken from the HTML body or a single query argument.
 
 ===== As multiple query parameters
 
 Parsing a query transported component-wise (as three independent query arguments) requires re-assembly:
+//TODO ? Those three independent query arguments: it must be three? It is three in the example below but could be any number?
```
Comment (aniluede) [inline documentation/modules/rql/pages/developer-guide.adoc:82]: just example

Comment (aniluede) [reply documentation/modules/rql/pages/developer-guide.adoc:82]: best practice 1 oder 3 (select filter option)

File: documentation/modules/rql/pages/developer-guide.adoc [hunk 12]
```diff
@@ -225,45 +235,48 @@ Not yet supported.
...
+* Logical: `and`, `or` and `not`
 
-Top-level properties, nested properties and collections of properties can be referenced in query expressions.
+Top-level properties, nested properties, and collections of properties can be referenced in query expressions.
 In the following example, filters are defined on properties of a `Person` entity.
 A Person has a `firstName` top-level property, a property `addresses` that references a subentity of type `Address` and a list of `Hobby` entities.
+//TODO ? Does capitalization matter here? (Doesn't match with the example. Plus wondering why Hobby is not hobby if first levels seemed to be lower case.)
 
 ----
 filter=eq(firstName,"John")
```
Comment (aniluede) [inline documentation/modules/rql/pages/developer-guide.adoc:247]: examples with slashes should be dots

File: documentation/modules/rql/pages/developer-guide.adoc [hunk 13]
```diff
@@ -5,10 +5,12 @@
 == RQL Parser
 
 This Maven module contains the antlr4 grammar as well as the parser to create the _RqlQueryModel_ from a string or vice versa.
+//TODO ? To which degree is the antlr4 grammar a known thing? Any external link that could be added?
```
[the remark this answers is not in the document]
Comment (aniluede) [reply documentation/modules/rql/pages/developer-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/f9e9f98b12f2a110cfcd6695c4b1e2f0c73f5343

File: documentation/modules/rql/pages/user-guide.adoc (hunk 3 above)
Comment (aniluede) [reply documentation/modules/rql/pages/user-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/2192784ca4ef47774b167c992259c2fc5b2632d5

File: documentation/modules/rql/pages/user-guide.adoc (hunk 4 above)
Comment (aniluede) [reply documentation/modules/rql/pages/user-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/f471ec5aca70d6cfbc04b2d9a757ac6979aa3cd3

File: documentation/modules/rql/pages/user-guide.adoc (hunk 5 above)
Comment (aniluede) [reply documentation/modules/rql/pages/user-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/3d590934c08a8b1b15cb265e8b785059a182772b

File: documentation/modules/rql/pages/user-guide.adoc (hunk 6 above)
Comment (aniluede) [reply documentation/modules/rql/pages/user-guide.adoc]: Done

https://github.com/bci-oss/semanticstack-rql/pull/69/commits/b3ccbfce0461cc81e2a560e8c7617827a07e2f21

https://github.com/bci-oss/semanticstack-rql/pull/69/commits/570101d4facef45d549da978e53e599913e0eda6

File: documentation/modules/rql/pages/user-guide.adoc (hunk 7 above)
Comment (aniluede) [reply documentation/modules/rql/pages/user-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/1ebe646bb30b87ff32516e5b43cf6d3375910656

File: documentation/modules/rql/pages/user-guide.adoc (hunk 8 above)
Comment (aniluede) [reply documentation/modules/rql/pages/user-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/1e458970200ea626548df407c0e4249529893341

File: documentation/modules/rql/pages/user-guide.adoc (hunk 9 above)
Comment (aniluede) [reply documentation/modules/rql/pages/user-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/1e458970200ea626548df407c0e4249529893341

File: documentation/modules/rql/pages/developer-guide.adoc (hunk 10 above)
Comment (aniluede) [reply documentation/modules/rql/pages/developer-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/aa243f00cc7d0f034db146dde9b026986b5caa30

Comment (aniluede) [reply documentation/modules/rql/pages/developer-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/aa243f00cc7d0f034db146dde9b026986b5caa30

File: documentation/modules/rql/pages/developer-guide.adoc (hunk 11 above)
Comment (aniluede) [reply documentation/modules/rql/pages/developer-guide.adoc]: Done -- https://github.com/bci-oss/semanticstack-rql/pull/69/commits/c476711ba83f9ef6b06189d6c36ef4a5a979c5f0

File: documentation/modules/rql/pages/developer-guide.adoc (hunk 12 above)
Comment (aniluede) [reply documentation/modules/rql/pages/developer-guide.adoc:239]: Duplicate entry, has been mentioned in a previous comment, is done

**Review by MelleD — APPROVED** [review]

**Review by MelleD — APPROVED** [review]

**Pull request merged**